vision_ocv/benchmark/modules/src/BM_ocv_vision_factory_test.cpp

```diff
@@ -0,0 +1,175 @@
+// Copyright 2023 Klepsydra Technologies AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "config.h"
+#include <benchmark/benchmark.h>
+#include <klepsydra/core/event_emitter_middleware_provider.h>
+#include <klepsydra/high_performance/event_loop_middleware_provider.h>
+#include <klepsydra/vision_ocv/file_image_stream_service.h>
+#include <klepsydra/vision_ocv/image_data_factory.h>
+#include <opencv2/opencv.hpp>
+#include <spdlog/spdlog.h>
+class FactoryPerformanceTest : public benchmark::Fixture
+{
+public:
+    FactoryPerformanceTest()
+        : benchmark::Fixture()
+    {}
+    ~FactoryPerformanceTest() {}
+
+    void SetUp(const ::benchmark::State &state)
+    {
+        int poolSize = state.range(0);
+        bool useEventLoop = state.range(1), withFactory = poolSize == 0;
+        width = height = state.range(2);
+
+        eventsListened = 0;
+        if (withFactory) {
+            _imageDataFactory.reset(
+                new kpsr::vision_ocv::ImageDataFactory(width,
+                                                       height,
+                                                       cv::IMREAD_REDUCED_GRAYSCALE_2,
+                                                       "raw_image"));
+            buildFunction = _imageDataFactory->initializerFunction;
+            _cloneFunction = _imageDataFactory->eventClonerFunction;
+        } else {
+            buildFunction = [](kpsr::vision_ocv::ImageData &src) {
+                src = kpsr::vision_ocv::ImageData(src);
+            };
```

Comment:
just use `buildFunction = nullptr` to make it use the default constructor. Otherwise this looks very confusing.

Reply:
I suppose cloneFunction should not be nullptr, since [here](https://github.com/klepsydra-technologies/kpsr-core/blob/main/core/modules/core/include/klepsydra/core/object_pool_publisher.h#L73), for example, assign operator is used instead of calling clone function and ImageData does not override said operator.

Reply:
Correct. that's why only the buildFunction should be nullptr.